docs/DeveloperGuide.md

```diff
@@ -1,38 +1,619 @@
-# Developer Guide
+# Apollo Developer Guide
+
+## Table of Contents
```

Comment:
I like that the DG is so detailed! However, there might be too much text as a result. Perhaps you could better create separate indentations and illustrations such as flowcharts?